contracts/src/RealitioForeignArbitrationProxyWithAppeals.sol

```diff
@@ -0,0 +1,604 @@
+        return round.hasPaid[_answer];
+    }
+
+    /** @notice Sends the fee stake rewards and reimbursements proportional to the contributions made to the winner of a dispute. Reimburses contributions if there is no winner.
+     *  @param _arbitrationID The ID of the arbitration.
+     *  @param _beneficiary The address to send reward to.
+     *  @param _round The round from which to withdraw.
+     *  @param _answer The answer to query reward from.
+     *  @return reward The withdrawn amount.
+     */
+    function withdrawFeesAndRewards(
+        uint256 _arbitrationID,
+        address payable _beneficiary,
+        uint256 _round,
+        uint256 _answer
+    ) public override returns (uint256 reward) {
+        Arbitration storage arbitration = arbitrations[_arbitrationID];
+        Round storage round = arbitration.rounds[_round];
+        require(arbitration.status == Status.Ruled, "Dispute not resolved");
+        uint256 finalAnswer = uint256(arbitration.answer);
+        // Allow to reimburse if funding of the round was unsuccessful.
+        if (!round.hasPaid[_answer]) {
+            reward = round.contributions[_beneficiary][_answer];
+        } else if (finalAnswer == 0 || !round.hasPaid[finalAnswer]) {
+            // Reimburse unspent fees proportionally if there is no winner and loser. Also applies to the situation where the ultimate winner didn't pay appeal fees fully.
+            reward = round.fundedAnswers.length > 1
+                ? (round.contributions[_beneficiary][_answer] * round.feeRewards) /
+                    (round.paidFees[round.fundedAnswers[0]] + round.paidFees[round.fundedAnswers[1]])
+                : 0;
+        } else if (finalAnswer == _answer) {
+            // Reward the winner.
+            reward = round.paidFees[_answer] > 0
+                ? (round.contributions[_beneficiary][_answer] * round.feeRewards) / round.paidFees[_answer]
+                : 0;
+        }
+
+        if (reward != 0) {
+            round.contributions[_beneficiary][_answer] = 0;
+            _beneficiary.transfer(reward);
+            emit Withdrawal(_arbitrationID, _round, _answer + 1, _beneficiary, reward);
+        }
+    }
+
+    /** @notice Allows to withdraw any reimbursable fees or rewards after the dispute gets solved for multiple ruling options (answers) at once.
+     *  @param _arbitrationID The ID of the arbitration.
+     *  @param _beneficiary The address that made contributions.
+     *  @param _round The round from which to withdraw.
+     *  @param _contributedTo Answers that received contributions from contributor.
+     */
+    function withdrawFeesAndRewardsForMultipleRulings(
+        uint256 _arbitrationID,
+        address payable _beneficiary,
+        uint256 _round,
+        uint256[] memory _contributedTo
+    ) public override {
+        for (uint256 contributionNumber = 0; contributionNumber < _contributedTo.length; contributionNumber++) {
```

Comment:
Wouldn't this make this function `O(n)`, where `n` is the number of answers the beneficiary has contributed to?

Reply:
True, I'll put the comment, and in the next function as well